### PR TITLE
fix(crap4ts): catch upstream oxc parser panic on malformed input

### PR DESCRIPTION
**Pull request by cmbays**
## Summary

The coverage-guided nightly fuzz lane (`fuzz_oxc_walker`, ~971k execs on first fire) found a **real panic** in the crap4ts oxc walker on adversarial malformed input. This fixes it and pins it with a committed regression seed.

## The bug

oxc 0.129's tuple-type **error recovery** is the culprit, entirely upstream and inside `Parser::parse`:

```
parse_tuple_type
  → diagnostics::required_element_cannot_follow_optional_element   (a required element after an optional one, e.g. `[fff?,`)
  → Span::label → miette SourceSpan::from → Span::size
  → assertion failed: self.start <= self.end          ❌ panic
```

On malformed input oxc builds an **inverted diagnostic span** (`start > end`) and `Span::size` asserts. It happens before any AST reaches our walker, so the walker's own (already saturating, panic-safe) span→line conversion never gets a chance.

Minimized 18-byte crasher: `Z<\x0b?[fff?,\r\r\r\r\r\r\r\r`.

## The fix

The `ComplexityPort` contract is explicit: **arbitrary bytes yield `Ok`/`Err`, never a panic** (the `fuzz_oxc_walker` target encodes this). The walker now wraps `Parser::parse` in `catch_unwind` and routes a caught panic through the **same** `CrapError::SourceParse` path a normal parse failure already takes. `AssertUnwindSafe` is sound here — on the panic path the arena + any half-built AST are dropped untouched, so no broken state is observed. A real user analyzing a syntactically-broken `.ts` file gets a clean "could not parse" error instead of an abort. Removable once the upstream oxc fix lands and the pin is bumped past it.

## Regression net (two layers)

- `tests/fuzz_oxc_walker/crashes/crash-oxc-inverted-tuple-span` — the minimized crasher, committed; the stable per-PR `replay_committed_corpus_and_crashes` test drives it forever (no nightly required).
- `walker_smoke::upstream_oxc_parser_panic_is_caught_as_source_parse` — a focused unit regression asserting graceful `SourceParse` at the code site.

## Gates (local)

- `cargo fmt --check` ✓ · `cargo +1.96.0 clippy -p crap4ts -p crap-core --all-targets -D warnings` ✓
- `cargo nextest run --workspace --all-targets` → **1350 passed** (incl. new test, crash-seed replay, all bolero targets) ✓
- `cargo +1.93 check -p crap4ts --all-targets` (MSRV) ✓ — std-only change, no new deps
- **Strict-8 production gate** (`crap.toml`, cognitive, 3 prod roots): PASS — `OxcWalker::extract` CC 5 / 100% cov, worst-in-repo 8.0, 0 above threshold

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Enhanced error handling to prevent application crashes when encountering malformed input; parser failures are now gracefully converted into clear, human-readable error messages.

* **Tests**
  * Added regression tests to verify robust error handling for edge cases with problematic source code.

### CODE REVIEW

File: crates/crap4ts/src/adapters/walker/mod.rs [hunk 1]
```diff
@@ -145,7 +145,31 @@ impl ComplexityPort for OxcWalker {
...
+            Parser::new(&allocator, source, source_type).parse()
+        }));
+        let ret = match parsed {
+            Ok(ret) => ret,
+            Err(_) => {
+                return Err(CrapError::SourceParse(format!(
+                    "{file_path}: could not parse (malformed source)"
+                )));
+            }
+        };
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap4ts/src/adapters/walker/mod.rs:176]: Agreed — surfaced the panic payload. The error now downcasts the caught `Box<dyn Any>` (`&str`/`String`) and includes it: `internal parser error on malformed source: <reason>` (commit d70b003). A future *new* upstream parser-panic class is now diagnosable from the error text instead of being flattened into one opaque message.

Keeping the catch-all (rather than matching specific panic shapes) is deliberate: the `ComplexityPort` contract is that arbitrary bytes yield `Ok`/`Err` and never a panic, so any upstream panic must be handled uniformly. The walker's own code (AST traversal, span→line conversion) remains panic-checkable on the `Ok` path.

**Comment by cmbays** [comment]
@coderabbitai review
